packages/@react-aria/list/src/useList.ts

```diff
@@ -0,0 +1,192 @@
+ * A list displays data in a single columns and enables a user to navigate its contents via directional navigation keys.
+ * @param props - Props for the list.
+ * @param state - State for the list, as returned by `useListState`.
+ * @param ref - The ref attached to the list element.
+ */
+export function useList<T>(props: AriaListOptions<T>, state: ListState<T>, ref: RefObject<HTMLElement>): ListViewAria {
+  // Rough copy of useGrid, but modifications + things removed for ListView specific case
+  let {
+    isVirtualized,
+    keyboardDelegate,
+    getRowText = (key) => state.collection.getItem(key)?.textValue,
+    onAction
+  } = props;
+  let formatMessage = useMessageFormatter(intlMessages);
+
+  if (!props['aria-label'] && !props['aria-labelledby']) {
+    console.warn('An aria-label or aria-labelledby prop is required for accessibility.');
+  }
+
+  let {listProps} = useSelectableList({
+    selectionManager: state.selectionManager,
+    collection: state.collection,
+    disabledKeys: state.disabledKeys,
+    ref,
+    keyboardDelegate: keyboardDelegate,
+    isVirtualized,
+    selectOnFocus: state.selectionManager.selectionBehavior === 'replace'
+  });
+
+  let id = useId();
+  listMap.set(state, {id, onAction});
+
+  // This is useHighlightSelectionDescription copy pasted, it isn't exposed by react-aria/grid.
+  let modality = useInteractionModality();
+  // null is the default if the user hasn't interacted with the list at all yet or the rest of the page
+  let shouldLongPress = (modality === 'pointer' || modality === 'virtual' || modality == null)
+    && typeof window !== 'undefined' && 'ontouchstart' in window;
+
+  let interactionDescription = useMemo(() => {
+    let selectionMode = state.selectionManager.selectionMode;
+    let selectionBehavior = state.selectionManager.selectionBehavior;
+
+    let message = undefined;
+    if (shouldLongPress) {
+      message = formatMessage('longPressToSelect');
+    }
+
+    return selectionBehavior === 'replace' && selectionMode !== 'none' && onAction ? message : undefined;
+  }, [state.selectionManager.selectionMode, state.selectionManager.selectionBehavior, onAction, formatMessage, shouldLongPress]);
+
+  let descriptionProps = useDescription(interactionDescription);
+
+  let domProps = filterDOMProps(props, {labelable: true});
+  let gridProps: HTMLAttributes<HTMLElement> = mergeProps(
+    domProps,
+    {
+      role: 'grid',
+      id,
+      'aria-multiselectable': state.selectionManager.selectionMode === 'multiple' ? 'true' : undefined
+    },
+    listProps,
+    descriptionProps
+  );
+
+  if (isVirtualized) {
+    gridProps['aria-rowcount'] = state.collection.size;
+    gridProps['aria-colcount'] = 1;
+  }
+
+  // Many screen readers do not announce when items in a grid are selected/deselected.
+  // We do this using an ARIA live region.
+  let selection = state.selectionManager.rawSelection;
+  let lastSelection = useRef(selection);
+  useUpdateEffect(() => {
```

Comment:
Maybe a candidate for logic to be split out into a separate hook outside the grid package in the future? Will have to see to investigate if this issue is grid specific. Comes from useGrid